plugin/trino-hive/src/test/java/io/trino/plugin/hive/coercions/TestDateCoercer.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive.coercions;
+
+import io.trino.spi.type.DateType;
+import io.trino.spi.type.Type;
+import io.trino.spi.type.VarcharType;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.time.LocalDate;
+
+import static io.airlift.slice.Slices.utf8Slice;
+import static io.trino.plugin.hive.HiveTimestampPrecision.NANOSECONDS;
+import static io.trino.plugin.hive.coercions.TestTimestampCoercer.assertCoercions;
+
+public class TestDateCoercer
+{
+    @DataProvider(name = "validDateProvider")
+    public static Object[][] validDateProvider()
+    {
+        return new Object[][] {{"2023-04-13"}, {"+10000-04-13"}, {"0221-03-14"}, {"-1323-01-17"}};
```

Comment:
Can we move each arguments to a new line - so tomorrow if we are planning to add additional test cases the diff would be easy.